sql/hive/src/test/scala/org/apache/spark/sql/sources/BucketedReadSuite.scala

```diff
@@ -59,6 +60,77 @@ class BucketedReadSuite extends QueryTest with SQLTestUtils with TestHiveSinglet
     }
   }
 
+  test("read partitioning bucketed tables with bucket pruning filters") {
```

Comment:
we also need a low-level test like `read bucketed data` to check the RDD partitions for un-needed bucket id is empty.